src/libtiled/varianttomapconverter.cpp

```diff
@@ -184,6 +185,20 @@ Properties VariantToMapConverter::toProperties(const QVariant &propertiesVariant
     return properties;
 }
 
+Components VariantToMapConverter::toComponents(const QVariant &componentsVariant) const
+{
+    Components components;
+
+    for (QVariant component : componentsVariant.toList()) {
+        QVariantMap componentMap = component.toMap();
```

Comment:
```suggestion
        const QVariantMap componentMap = component.toMap();
```
It is important to declare this map as `const`, otherwise `componentMap[QStringLiteral("name")]` will cause a detach.